File: design-pattern-jvm/app/src/main/java/design/pattern/java/patterns/creational/singleton/DBConnection.java
package design.pattern.java.patterns.creational.singleton;

public class DBConnection {

  private static DBConnection instance;
  private String host;
  private Number port;
  private String username;
  private String password;

  
  private DBConnection(String host,Number port, String username, String password) {
    super();
    this.host = host;
    this.port = port;
    this.username = username;
    this.password = password;
  }
  public static DBConnection create(String host, Number port, String username, String password) {
    if (instance == null) {
      DBConnection.instance = new DBConnection(host, port, username, password);
      return DBConnection.instance;
    }
    return DBConnection.instance;
  } 
  
  public String getHost() {
    return this.host;
  }
  public Number getPort() {
    return this.port;
  }
  public String getUsername() {
    return this.username;
  }
  public String getPassword() {
    return this.password;
  }
}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/creational/singleton/DBConnection.kt
package design.pattern.kotlin.patterns.creational.singleton

class DBConnection private constructor(
    val host: String,
    val port: Number,
    val username: String,
    val password: String
) {

    companion object {
        private var instance: DBConnection? = null
        fun create(host: String, port: Number, username: String, password: String): DBConnection? {
            if (instance == null) {
                instance = DBConnection(host, port, username, password)
                return instance
            }
            return instance
        }
    }
}